### PR TITLE
[Reward] Set reward objects to backend

### PR DESCRIPTION
**Pull request by laggu**
## Proposed changes

This pr is a part of Reward Refactoring
- Set RewardDistributor and StakingManager to CN and istanbul backend
- make a function GetItemAtNumberByIntKey() to governance
This is for not accessing governance package directly. 
Keys of item are the constant defined in params package.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

The next pr will replace current code to the refactoring code

### CODE REVIEW

File: reward/staking_manager.go [hunk 1]
```diff
@@ -108,3 +108,7 @@ func (sm *stakingManager) handleChainHeadEvent() {
 		}
 	}
 }
+
```
Comment (KimKyungup) [inline reward/staking_manager.go]:
```suggestion

// UnSubscribe can unsubscribe a subscription to listen chain head event..
```

Comment (KimKyungup) [reply reward/staking_manager.go]: Do you have any reason for writing `UnSubscribe` instead of `Unsubscribe`?

File: governance/handler.go [hunk 2]
```diff
@@ -422,3 +422,7 @@ func (gov *Governance) GetGovernanceItemAtNumber(num uint64, key string) (interf
 		return nil, ErrItemNotFound
 	}
 }
+
+func (gov *Governance) GetItemAtNumberByIntKey(num uint64, key int) (interface{}, error) {
```
Comment (KimKyungup) [inline governance/handler.go]: How about this?
```suggestion
func (gov *Governance) GetItemAtBlockNumberByKey(num uint64, key int) (interface{}, error) {
```

File: consensus/istanbul/backend/backend.go [hunk 3]
```diff
@@ -121,6 +123,9 @@ type backend struct {
 	// Last Block Number which has current Governance Config
 	lastGovernanceBlock uint64
 
+	rewardDistributor *reward.RewardDistributor
```
Comment (KimKyungup) [inline consensus/istanbul/backend/backend.go]: `rewardDistributor` is cool. :)

File: reward/staking_manager.go (hunk 1 above)
Comment (laggu) [reply reward/staking_manager.go]: I've changed it..

File: reward/staking_manager.go [hunk 4]
```diff
@@ -108,3 +108,8 @@ func (sm *stakingManager) handleChainHeadEvent() {
...
+// UnSubscribe can unsubscribe a subscription to listen chain head event.
```
Comment (KimKyungup) [inline reward/staking_manager.go]:
```suggestion
// Unsubscribe can unsubscribe a subscription to listen chain head event.
```

File: node/cn/backend.go [hunk 5]
```diff
@@ -528,12 +535,18 @@ func (s *CN) Start(srvr p2p.Server) error {
 	if s.lesServer != nil {
 		s.lesServer.Start(srvr)
 	}
+	if s.governance.ProposerPolicy() == uint64(istanbul.WeightedRandom) {
```
Comment (jeonghonah) [inline node/cn/backend.go]: Is there any need for a nil check of the `s.governance`?

Comment (laggu) [reply node/cn/backend.go]: No, there is no need. CN is created with governance

File: reward/staking_manager.go [hunk 6]
```diff
@@ -108,3 +108,15 @@ func (sm *stakingManager) handleChainHeadEvent() {
 		}
 	}
 }
+
+// Unsubscribe can unsubscribe a subscription to listen chain head event.
+func (sm *StakingManager) Unsubscribe() {
+	sm.chainHeadSub.Unsubscribe()
+}
+
+func (sm *StakingManager) IsSubscribing() bool {
```
Comment (jeonghonah) [inline reward/staking_manager.go]:
```suggestion
func (sm *StakingManager) IsSubscribing() bool {
	return sm.chainHeadSub != nil
}
```

Comment (laggu) [reply reward/staking_manager.go]: reflect

File: governance/handler.go (hunk 2 above)
Comment (jeonghonah) [inline governance/handler.go]: Description comment is missing.

File: node/cn/backend.go [hunk 7]
```diff
@@ -99,7 +104,8 @@ type CN struct {
 
 	components []interface{}
 
-	governance *governance.Governance
+	governance     *governance.Governance
+	stakingManager *reward.StakingManager
 }
 
 func (s *CN) AddLesServer(ls LesServer) {
```
Comment (jeonghonah) [inline node/cn/backend.go]: Method description comment is missing.

File: reward/reward_distributor.go [hunk 8]
```diff
@@ -43,14 +42,12 @@ func isEmptyAddress(addr common.Address) bool {
 }
 
 type RewardDistributor struct {
-	sm  *stakingManager
 	rcc *rewardConfigCache
 	gh  governanceHelper
 }
 
-func NewRewardDistributor(bc *blockchain.BlockChain, gh governanceHelper) *RewardDistributor {
+func NewRewardDistributor(gh governanceHelper) *RewardDistributor {
```
Comment (jeonghonah) [inline reward/reward_distributor.go]: Method description comment is missing.

File: reward/staking_manager.go [hunk 9]
```diff
@@ -46,7 +46,7 @@ func newStakingManager(bc *blockchain.BlockChain, gh governanceHelper) *stakingM
 }
 
 // GetStakingInfoFromStakingCache returns a corresponding stakingInfo for a blockNum.
```
Comment (jeonghonah) [inline reward/staking_manager.go]:
```suggestion
// getStakingInfo returns a corresponding stakingInfo for a blockNum.
```

File: reward/staking_manager.go [hunk 10]
```diff
@@ -78,14 +78,14 @@ func (sm *stakingManager) updateStakingCache(blockNum uint64) (*StakingInfo, err
...
-func (sm *stakingManager) subscribe() {
+func (sm *StakingManager) Subscribe() {
 	sm.chainHeadSub = sm.bc.SubscribeChainHeadEvent(sm.chainHeadCh)
 
 	go sm.handleChainHeadEvent()
 }
 
-func (sm *stakingManager) handleChainHeadEvent() {
-	defer sm.chainHeadSub.Unsubscribe()
+func (sm *StakingManager) handleChainHeadEvent() {
```
Comment (jeonghonah) [inline reward/staking_manager.go]: Method description comment is missing.

File: reward/staking_manager.go [hunk 11]
```diff
@@ -108,3 +108,12 @@ func (sm *stakingManager) handleChainHeadEvent() {
 		}
 	}
 }
+
+// Unsubscribe can unsubscribe a subscription to listen chain head event.
+func (sm *StakingManager) Unsubscribe() {
+	sm.chainHeadSub.Unsubscribe()
+}
+
+func (sm *StakingManager) IsSubscribing() bool {
```
Comment (jeonghonah) [inline reward/staking_manager.go]: Method description comment is missing.

File: governance/handler.go (hunk 2 above)
Comment (laggu) [reply governance/handler.go]: thanks for your comment. I'll do it in a next pr

File: node/cn/backend.go (hunk 7 above)
Comment (laggu) [reply node/cn/backend.go]: I'll make a new pr for method description comments

File: reward/reward_distributor.go (hunk 8 above)
Comment (laggu) [reply reward/reward_distributor.go]: I'll make a new pr for method description comments

File: reward/staking_manager.go (hunk 9 above)
Comment (laggu) [reply reward/staking_manager.go]: I'll make a new pr for method description comments

File: reward/staking_manager.go (hunk 10 above)
Comment (laggu) [reply reward/staking_manager.go]: I'll make a new pr for method description comments

File: reward/staking_manager.go (hunk 11 above)
Comment (laggu) [reply reward/staking_manager.go]: I'll make a new pr for method description comments

File: reward/staking_manager.go (hunk 9 above)
Comment (jeonghonah) [reply reward/staking_manager.go]: This is a typo. Let's fix it in the next PR.

**Pull request merged**